googleapiclient/discovery_cache/__init__.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Caching utility for the discovery document."""
+
+
+from __future__ import absolute_import
+
+import logging
+import datetime
+
+
+DISCOVERY_DOC_MAX_AGE = 60 * 60 * 24  # 1 day
+
+
+def autodetect():
+  """Detects an appropriate cache module and returns it.
+
+  Returns:
+    An object with the following methods; get(url), set(url, content).
+  """
+  try:
+    from google.appengine.api import memcache
+    from . import appengine_memcache
+    return appengine_memcache.Cache(max_age = DISCOVERY_DOC_MAX_AGE)
+  except:
+    try:
+      from . import file_cache
+      return file_cache.Cache(max_age = DISCOVERY_DOC_MAX_AGE)
```

Comment:
Drop the whitespace from either side of the equals sign here. When passing parameters by keyword the controlling style says to never use spaces on either side.